### ISSUE TITLE
Single text prediction changes.

### ISSUE THREAD

**Issue by anshkumar** [QwenLM/Qwen2-VL#573]
Hi I wanted to know what changes do I need to do if I only want to predict a single text as output for any input text+image. 

My current training settings:

```
accelerate launch --fp8_backend=msamp --fp8_opt_level=O2 src/training/train.py \
    --deepspeed scripts/zero3.json \
    --optim adamw_bnb_8bit \
    --model_id $MODEL_NAME \
    --data_path vlm_dataset.json \
    --image_folder vlm_dataset \
    --lora_enable False \
    --tune_merger True \
    --freeze_vision_tower False \
    --freeze_llm False \
    --bf16 True \
    --output_dir output \
    --num_train_epochs 10 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 32 \
    --min_pixels $((49 * 28 * 28)) \
    --max_pixels $((49 * 28 * 28)) \
    --learning_rate 1e-5 \
    --merger_lr 1e-5 \
    --vision_lr 2e-6 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --report_to tensorboard \
    --lazy_preprocess True \
    --save_strategy "epoch" \
    --save_total_limit 1 \
    --dataloader_num_workers 4
```

Training accuracy is eventually decreases to zero, but inferencing results are not good.

<img width="1074" alt="Screenshot 2024-12-02 at 12 53 05 PM" src="https://github.com/user-attachments/assets/ae380041-3392-4911-bb1d-f6a7bba4f0af">
